akka-cluster/src/main/scala/akka/cluster/ClusterReadView.scala

```diff
@@ -45,17 +48,16 @@ private[akka] class ClusterReadView(cluster: Cluster) extends Closeable {
   @volatile
   private var _latestStats = CurrentInternalStats(GossipStats(), VectorClockStats())
 
-  val selfAddress = cluster.selfAddress
+  val selfAddress: Address = cluster.selfAddress
 
   // create actor that subscribes to the cluster eventBus to update current read view state
   private val eventBusListener: ActorRef = {
     cluster.system
       .systemActorOf(Props(new Actor with RequiresMessageQueue[UnboundedMessageQueueSemantics] {
         override def preStart(): Unit = cluster.subscribe(self, classOf[ClusterDomainEvent])
-        override def postStop(): Unit = cluster.unsubscribe(self)
 
         def receive: Receive = {
-          case e: ClusterDomainEvent =>
+          case e: ClusterDomainEvent if !_closed =>
```

Comment:
This actor is stopped via PoisonPill from `close()` and it was probably rather arbitrary what events it could receive during the shutdown process. Now it ignores everything after close, i.e. `state` will be latest known `CurrentClusterState` before shutdown. Alternative could be to clear everything from close.